### ISSUE TITLE
[Doc]: Development workflow doc has lots of typos and clunky sentences

### ISSUE THREAD

**Issue by saranti** [matplotlib/matplotlib#24762]
### Documentation Link

https://matplotlib.org/devdocs/devel/development_workflow.html

### Problem

There are typos and/or grammer errors in the following lines on this page at https://github.com/matplotlib/matplotlib/blob/main/doc/devel/development_workflow.rst:

```
59 Making an new branch for each set of related changes will make it easier for

150 Enter a title for the set of changes, and some explanation of what you've done.
151 Say if there is anything you'd like particular attention for - like a

213 There's an embarrassing typo in a commit you made? Or perhaps the you
214 made several false starts you would like the posterity not to see.

282 and the history looks now like this::

317 worry that ``main`` has got a little ahead.  But sometimes, the changes in

352 When all looks good you can delete your backup branch::

393 fail with an error that looks like ::

405 If this push had succeed then the commits ``A``, ``B``, and ``C`` would no
406 longer be referenced by any branch and be discarded:

420 versions from our local branch.  In this case, what we want to do is ::

431 history and if anyone has fetched the old commits will have a different view
432 of history which can cause confusion.
```

and possibly others that I missed.

### Suggested improvement

Fix typos

**Comment by saranti** [comment]
Additionally, the section [Recovering from mess-ups ](https://matplotlib.org/devdocs/devel/development_workflow.html#recovering-from-mess-ups) looks like it should come after the [Rewriting commit history ](https://matplotlib.org/devdocs/devel/development_workflow.html#rewriting-commit-history) section. (Learning how to make a save point should preceed the section on how to restore a save point ?.)

**Comment by timhoffm** [comment]
Thank you for the suggestions! Do you want to turn them into a pull request?

**Comment by saranti** [comment]
> Thank you for the suggestions! Do you want to turn them into a pull request?

I thought this could be a good first issue for someone. What do you think?

**Comment by luke396** [comment]
Hello, I am new to contributing to open source projects and I saw that there is a spelling correction issue that I would like to take on. Is it okay for a newcomer like me to work on this issue? I am excited to contribute and learn from the community.

**Comment by saranti** [comment]
> Hello, I am new to contributing to open source projects and I saw that there is a spelling correction issue that I would like to take on. Is it okay for a newcomer like me to work on this issue? I am excited to contribute and learn from the community.

Hi Luke. It looks like somebody already beat you to it.

**Comment by luke396** [comment]
> > Hello, I am new to contributing to open source projects and I saw that there is a spelling correction issue that I would like to take on. Is it okay for a newcomer like me to work on this issue? I am excited to contribute and learn from the community.
> 
> Hi Luke. It looks like somebody already beat you to it.

Yes, I noticed that too 😿 , my PR is still in progress.

**Comment by melissawm** [comment]
@luke396 feel free to browse the repo and look at the ["good first issue" label](https://github.com/matplotlib/matplotlib/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Good+first+issue%22) to find other stuff to work on. Cheers!

**Comment by QuLogic** [comment]
I'm going to assume this is fully fixed by #24782.
